filebeat/fileset/modules.go

```diff
@@ -392,6 +392,11 @@ func checkAvailableProcessors(esClient PipelineLoader, requiredProcessors []Proc
 
 // LoadML loads the machine-learning configurations into Elasticsearch, if X-Pack is available
 func (reg *ModuleRegistry) LoadML(esClient PipelineLoader) error {
+	if !mlimporter.IsCompatible(esClient) {
```

Comment:
Would it be also an option to return an error instead? Like this we make sure the user is aware of the change and can adjust the code.

Reply:
What would be the expected action of the user? Change the `setup` command to exclude loading ML jobs?

What I have in mind is returning an error message when the user tried to load ML with the flag `--machine-learning` specified. Otherwise, just let them know that ML jobs are no longer loaded and do not return an error. WDYT?

Reply:
If a user runs the setup command, are the machine learning jobs installed by default? If yes, agree we should break this behaviour. If it is only if a flag was set manually either through config for command line flags, we should likely tell the user about it by erroring out. What I worry is that users will not look at the logs and will be surprised at one point that nothing is loaded. At the same time this is not "system critical" I would expect.

Reply:
Yes, it is loaded by default. I will add an error message for the case when users set machine learning explicitly.